science-containers/Dockerfiles/firefly/cadc-sso/lib/src/main/java/org/opencadc/security/sso/TokenRelay.java

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * Provides classes and interfaces for handling Single Sign-On (SSO) authentication
+ * within the CADC (Canadian Astronomy Data Centre) environment.
+ *
+ * <p>This package includes the following key components:</p>
+ * <ul>
+ *   <li>{@link org.opencadc.security.sso.TokenRelay} - A class that retrieves the authentication token from the SSO cookie.</li>
+ * </ul>
+ *
+ * <p>The primary purpose of this package is to facilitate secure authentication and
+ * authorization mechanisms for accessing CADC services.</p>
+ */
+package org.opencadc.security.sso;
+
+import edu.caltech.ipac.firefly.data.userdata.UserInfo;
+import edu.caltech.ipac.firefly.server.RequestAgent;
+import edu.caltech.ipac.firefly.server.ServerContext;
+import edu.caltech.ipac.firefly.server.network.HttpServiceInput;
+import edu.caltech.ipac.firefly.server.security.SsoAdapter;
+import edu.caltech.ipac.firefly.server.util.Logger;
+
+import javax.servlet.http.Cookie;
+
+public class TokenRelay implements SsoAdapter {
```

Comment:
Add a Class comment with a brief description here, and possibly an example usage.

Reply:
Done.